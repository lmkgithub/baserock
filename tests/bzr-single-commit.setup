#!/bin/sh
#
# Creates a bzr repository with a single file and a single commit.
#
# Copyright (C) 2012  Codethink Limited
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.


set -e

# create the repository
repo="$DATADIR/bzr-test-repo"
mkdir "$repo"
cd "$repo"
bzr init --quiet trunk

# add the test file
cd "$repo/trunk"
echo "first line" > test.txt
bzr add --quiet test.txt

# make a commit
bzr commit --quiet -m "first commit"

# create the .lorry file for the repository
cat <<EOF > $DATADIR/bzr-test-repo.lorry
{
  "bzr-test-repo": {
    "type": "bzr",
    "url": "file://$repo/trunk"
  }
}
EOF


# create the working directory
test -d "$DATADIR/work-dir" || mkdir "$DATADIR/work-dir"
