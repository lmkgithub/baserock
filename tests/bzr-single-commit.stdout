refs/heads/trunk
first line
first commit
